### PR TITLE
[ltac2] Add Check Ltac2 to print only the type

### PR DESCRIPTION
**Pull request by tchajed**
**Kind:** feature

- [x] Added / updated test-suite
- [x] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).
- [ ] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

**Comment by tchajed** [comment]
@ppedrot I wanted to get a hang of Ltac2 so I figured I'd implement something.

This is in some ways half-baked, in that constructor arguments are not reported (although better than in `Print Ltac`, which doesn't even give the inductive type name).

Out of everything here my test is probably the most useful and I'd be happy to submit that separately (without the `Check Ltac2` calls of course).

File: user-contrib/Ltac2/tac2entries.ml [hunk 1]
```diff
@@ -869,6 +869,32 @@ end
...
+    let mkn =
+      try Some (Tac2env.locate_constructor qid)
+      with Not_found -> None
+    in
+    match mkn with
+    | Some kn ->
+      let data = Tac2env.interp_constructor kn in
+      let t = data.Tac2env.cdata_type in
+      Feedback.msg_notice (hov 2 (pr_qualid qid ++ spc () ++ str ":" ++ spc () ++ pr_typref t))
+    | None -> Feedback.msg_notice (hov 2 (pr_qualid qid ++ spc () ++ str ":" ++ spc () ++ str "Type"))
```
Comment (ppedrot) [inline user-contrib/Ltac2/tac2entries.ml]: This should probably be an error.

File: user-contrib/Ltac2/tac2entries.ml [hunk 2]
```diff
@@ -869,6 +869,32 @@ end
...
+      with Not_found -> user_err ?loc:qid.CAst.loc (str "Unknown tactic " ++ pr_qualid qid)
+    in
+    match kn with
+    | TacConstant kn ->
+      let data = Tac2env.interp_global kn in
+      let (_, t) = data.Tac2env.gdata_type in
+      let name = int_name () in
+      Feedback.msg_notice (hov 0 (pr_qualid qid ++ spc () ++ str ":" ++ spc () ++ pr_glbtype name t))
+    | TacAlias kn ->
+      Feedback.msg_notice (str "Alias to ...")
```
Comment (ppedrot) [inline user-contrib/Ltac2/tac2entries.ml]: You also need to fill this in.

**Review by ppedrot — COMMENTED** [review]
I am not sure about the usefulness of `Check` when there is already `Print` but YMMV.

File: user-contrib/Ltac2/tac2entries.ml (hunk 1 above)
Comment (tchajed) [reply user-contrib/Ltac2/tac2entries.ml]: This gets triggered by `Check Ltac2 ABC` where `ABC` is a type.

File: user-contrib/Ltac2/tac2entries.ml (hunk 2 above)
Comment (tchajed) [reply user-contrib/Ltac2/tac2entries.ml]: Printing doesn't do anything with it, so I don't know what to do.

File: user-contrib/Ltac2/tac2entries.ml (hunk 1 above)
Comment (ppedrot) [reply user-contrib/Ltac2/tac2entries.ml]: Oh, I didn't realize that uppercase types were allowed. This looks more like a bug than anything else to me :/ In any case, this shouldn't read "XYZ : Type" as there is no such thing as `Type` in the Ltac2 type system.

File: user-contrib/Ltac2/tac2entries.ml (hunk 2 above)
Comment (ppedrot) [reply user-contrib/Ltac2/tac2entries.ml]: WDYM? You could at least print the thing it aliases to.

Comment (JasonGross) [reply user-contrib/Ltac2/tac2entries.ml]: I read Tej as asking "how do I print things of this type?" or "how do I find the function that takes a thing of this type and makes it a Pp.t?"

**Review by Zimmi48 — APPROVED** [review]
Likely deserves a changelog entry.

**Comment by ppedrot** [comment]
I think we can close this PR, it seems to have died.

**Pull request closed without merging**